lisa/tools/tpm2.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from typing import Dict, List, Optional, Sequence, Union
+
+from lisa.executable import Tool
+from lisa.operating_system import CBLMariner
+from lisa.util import LisaException
+
+
+class Tpm2(Tool):
+    @property
+    def command(self) -> str:
+        return "tpm2"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def _install(self) -> bool:
+        if isinstance(self.node.os, CBLMariner):
+            self.node.os.install_packages("tpm2-tools")
+        else:
+            raise LisaException(
+                f"tool {self.command} can't be installed in distro {self.node.os.name}."
+            )
+        return self._check_exists()
+
+    def pcrread(
+        self, pcrs: Optional[Union[int, Sequence[int]]] = None
+    ) -> Dict[int, str]:
+        alg = "sha256"
+        pcrs = self._get_pcr_list(pcrs)
+        if len(pcrs) == 0:
+            pcrs_arg = "all"
+        else:
+            pcrs_arg = ",".join(map(str, pcrs))
+        cmd = f"pcrread {alg}:{pcrs_arg}"
+        cmd_result = self.run(
+            cmd,
+            expected_exit_code=0,
+            expected_exit_code_failure_message="failed to read PCR values",
+            shell=True,
+            sudo=True,
```

Comment:
If sudo is not necessary, don't use sudo by default.

Reply:
We need sudo to interact with the TPM device